clap_generate/src/man.rs

```diff
@@ -0,0 +1,287 @@
+use std::fmt::write;
+
+use clap::ArgSettings;
+
+/// Man page generator
+#[derive(Debug, Clone)]
+pub struct Man<'a> {
+    app: clap::App<'a>,
+    section: i8,
+    manual: Option<String>,
+    sections: Vec<(String, Vec<String>)>,
+}
+
+impl<'a> Man<'a> {
+    /// Create a new man page for the app.
+    pub fn create(
+        app: clap::App<'a>,
+        section: i8,
+        manual: Option<String>,
+        buf: &mut dyn std::io::Write,
+    ) {
+        let man = Man {
+            app,
+            section,
+            manual,
+            sections: Vec::new(),
+        };
+
+        man.write(buf)
+    }
+
+    /// Create a new builder for man pages.
+    pub fn new(app: clap::App<'a>, section: i8, manual: Option<String>) -> Self {
+        Man {
+            app,
+            section,
+            manual,
+            sections: Vec::new(),
+        }
+    }
+
+    /// Add a custom section to the man pages.
+    pub fn section(mut self, title: impl Into<String>, body: Vec<impl Into<String>>) -> Self {
+        self.sections
+            .push((title.into(), body.into_iter().map(|s| s.into()).collect()));
+        self
+    }
+
+    /// Write the manpage to a buffer.
+    pub fn write(self, buf: &mut dyn std::io::Write) {
+        let mut buffer = String::new();
```

Comment:
Why write to the `buffer` and then write to `buf` rather than just write directly to `buf`?

Reply:
Good point, I didn't really think about that :facepalm: